Services/CssClassHolder.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Collections.Generic;
+
+namespace Lombiq.BaseTheme.Services
+{
+    public class CssClassHolder : ICssClassHolder
+    {
+        private readonly Dictionary<string, HashSet<string>> _zones = new();
+
+        public ISet<string> Body { get; } = new HashSet<string>();
+
+        public void AddClassToZone(string zoneName, string className)
+        {
+            var classes = _zones.GetMaybe(zoneName);
+
+            if (classes == null)
+            {
+                classes = new();
+                _zones[zoneName] = classes;
+            }
+
+            classes.Add(className);
```

Comment:
Would you consider the following too convoluted?
```suggestion
           (_zones.GetMaybe(zoneName) ?? (_zones[zoneName] = new())).Add(className);
```

Reply:
I'm pretty sure the code analysis disallows assignment inside an expression. Let me check...

Reply:
Sorry for not checking myself. I don't have the solution open, currently. I just found the method body to be very verbose for what seems to be a simple implementation.

Reply:
It says:
```
CssClassHolder.cs(12,61): error S1121: Extract the assignment of '_zones[zoneName]' from this expression.
```
I agree, inline assigning is not good for readability.

The code analysis also prevents multiple assigning (eg. `_zones[zoneName] = classes = new();`) which could've let me collapse the `if` and make it into a simple three-liner. I think there is nothing wrong with that, but I don't write the rules. :)

Reply:
If trying to call something on a Dictionary value would be something more common, I could imagine an extension method like

```c#
public static TValue GetOrSet(this IDictionary<TKey, TValue> dict, TKey key, Func<TValue> valueInitializer)
{
    if (dict.TryGetValue(key, out var value)) return value;

    dict[key] = valueInitializer();

    return dict[key];
}
```

Reply:
Yeah, like in [ConcurrentDictionary.GetOrAdd](https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.getoradd?view=net-6.0). But I think it's pretty rare.

Reply:
Exactly 😀